src/main/java/frc/robot/commands/LimelightTravelDistanceCommand.java

```diff
@@ -0,0 +1,62 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants.LimelightConstants;
+import frc.robot.RobotContainer;
+import frc.robot.subsystems.DriveSubsystem;
+import frc.robot.subsystems.LimelightSubsystem;
+
+public class LimelightTravelDistanceCommand extends CommandBase{
+    private LimelightSubsystem m_LimelightSubsystem;
+    private DriveSubsystem m_DriveSubsystem;
+    private double speed;
+    private double travelDistance;
+    private double targetHeight;
+
+
+    public LimelightTravelDistanceCommand (double targetHeight) {
+        m_LimelightSubsystem = RobotContainer.limelightSubsystem;
+        m_DriveSubsystem = RobotContainer.driveSubsystem;
+        speed = 0;
+        this.targetHeight = targetHeight;
+        addRequirements(m_LimelightSubsystem, m_DriveSubsystem);
+    }
+    @Override
+    public void initialize() {
+
+    }
+
+    @Override
+    public void execute() {
+
+        travelDistance = m_LimelightSubsystem.getDepth(targetHeight) - LimelightConstants.kArmReach; // subtracting limelight's distance by claw's reach, subject to change
+
+        SmartDashboard.putNumber("Vertical Offset", m_LimelightSubsystem.getVerticalOffset());
+        SmartDashboard.putNumber("Depth to Target", m_LimelightSubsystem.getDepth(targetHeight));
+        SmartDashboard.updateValues();
+
+        double kPAlignDistance = 0.01;
+        speed = travelDistance * kPAlignDistance;
```

Comment:
You use `speed` in a couple places, but this is not actually the speed of the motors, but closer to the percent voltage provided to the motors.